current.go

```diff
@@ -61,32 +60,71 @@ func (r *VaultRole) Search(config SearchConfig) (Role, error) {
 		return Role{}, microerror.Maskf(notFoundError, "no vault secret at path '%s'", key.RoleName(config.ID, config.Organizations))
 	}
 
-	b, err := json.Marshal(secret.Data)
+	role, err := vaultSecretToRole(secret)
 	if err != nil {
 		return Role{}, microerror.Mask(err)
 	}
 
-	var internalRole role
-	err = json.Unmarshal(b, &internalRole)
-	if err != nil {
-		return Role{}, microerror.Mask(err)
+	role.ID = config.ID
+	return role, nil
+}
+
+// vaultSecretToRole makes required type casts / type checks and parsing to
+// extract role information from Vault api.Secret.
+func vaultSecretToRole(secret *api.Secret) (Role, error) {
+	var role Role
+
+	if allowBareDomains, ok := secret.Data["allow_bare_domains"].(bool); ok {
+		role.AllowBareDomains = allowBareDomains
+	} else {
+		return Role{}, microerror.Maskf(wrongTypeError, "Vault secret.Data[\"allow_bare_domains\"] type is %T, expected %T", secret.Data["allow_bare_domains"], allowBareDomains)
 	}
 
-	altNames := key.ToAltNames(internalRole.AllowedDomains)
-	organizations := key.ToOrganizations(internalRole.Organizations)
-	ttl, err := parseutil.ParseDurationSecond(internalRole.TTL)
-	if err != nil {
-		return Role{}, microerror.Mask(err)
+	if allowSubdomains, ok := secret.Data["allow_subdomains"].(bool); ok {
```

Comment:
IMO it is always good to separate logic and conditionals.

Reply:
Do you mean this should be written in following style:
```go
allowSubdomains, ok := secret.Data["allow_subdomains"]
if ok {
    // do the foo
} else {
   // do the bar
}
```

For most of the fields this would work, but for `allowed_domains` it doesn't work easily and I'd say using the present style fits here okayish. One alternative would be to introduce separate functions `toBool`, `toStringVal`, `toStringSlice`, `fromInterfaceSlicetoStringSlice`. That could make this specific function a bit more readable. WDYT?